### PR TITLE
Add some lab monsters

### PR DESCRIPTION
**Pull request by Venera3**
#### Summary
Content "More lab monsters"
#### Purpose of change

Shiny new labs need shiny new monsters, including some that can look after those elusive keycards. This PR adds a few alternate variants to evolved mutants, and two boss monsters for modular labs.

#### Describe the solution

Add five alternate variants to the experimental mutant's possible evolutions and a special boss version for a keycard drop. Also slightly adjusted  `tracking_distance` to allow for shorter follow ranges than 8.

TODO:
- [x] Add a custom arena to their designated spot, preferably as a nest to allow for variations down the line
- [x] Add a second boss for a different keycard
- [x] Some more playtesting

#### Describe alternatives you've considered

Just do some zombies, but those are boooring.
Not include them as normal variants for wild-type experimental mutants, but variety is nice.
Go ham on the harvest snippets as usual, but these are kind of the only thing holding back the modular labs so I wanted to get them out the door.

#### Testing
`KEEP_DISTANCE` monsters keep the new default distance. Monsters load, do their thing without incident. Combat testing as usual, plus pack-based testing since you're not supposed to run into them on their own.

#### Additional context
New mutant variants:

- Ungulate: big annoying lads with a stabby set of antlers and hit-and-run to better distract you with, they are generally very easily handled by even a midgame character in 1v4 but will get you a bit bloody while you take care of their friends.
- Arthropod: fast, with a venomous reaching stinger (damaging venom), and a debilitating-but-inaccurate webspinning attack that can seriously mess your day up if it hits and you're already weakened by the fight. Midgame characters still dispatch them one-on-one but they get banged up, four can ruin the day of even a lategame character.
- Bestial: Rebranded evolved mutant, big chunky melee attacker - hits like a truck and can seriously mess up midgame survivors. The impale attack got a slight nerf in that it doesn't down you automatically, but they can still hold their own in closed quarters
- Camouflaged: Hit-and-run dodgetanks with a painful bite and camouflage to mitigate ranged weapons somewhat, though without the per-debuffing venom of unseen hunters.
- Shelled: Armored tanks with a reaching tentacle snap and every grab attack in the book, they will make it much easier for the rest of the gang to tear your face off. Outdamaging their armor is a challenge, but bashing weapons or *big* guns do the trick fine enough.
- Transcendent: Generic version of the boss, squishy ranged buffer (ups speed, dodge and melee skill in range). If you let them pop their frenzy attack a few times the other mutants will tear you apart, though good luck catching them without a gun.

 <details>
<summary> Spoilers</summary>

Lab bosses:
- Mutant child: Probably an extremely lucky Alpha subject that got slightly broken by the experiments, and now that the lab fell they have all the time in the world to enhance their warped friends even further. Smaller, causes guilt on kill and drops a green science keycard plus a bunch of serums and usually some mutagen recipe books. 
- Melded Task Force (couldn't resist some acronym fun): A dissoluted devourer made up of lab security guys, accreted around some weird zombie master/necromancer type of figure. Hits hard and regenerates, parts of it fall off and become flesh raptors for extra fun. Needs a lot of killing to put down, and when he dies he drops a portal as a last "screw you". Oh, it also can open doors and follow you around. A lategame character (medieval weaponry, 7 skill, Brawling and survivor armor all around) has about 50% chance to outmelee them before running out of stamina aside from rapier-wielders who fare much better - that low stamina cost lets them poke till the sun goes down. Alternatively, some grenades work wonders.
</details>

### CODE REVIEW

**Review by actual-nh — COMMENTED** [review]
Just a few things, that I actually wrote a few hours ago and forgot to submit (got told that I can't reuse a bunch of stuff from this summer in the two online classes I'm teaching!).

File: data/json/monsters/mutant.json [hunk 1]
```diff
@@ -37,52 +37,252 @@
...
     "regen_morale": true,
     "flags": [ "SEES", "HEARS", "SMELLS", "KEENNOSE", "WARM", "BASHES" ]
   },
   {
     "id": "mon_mutant_evolved",
     "type": "MONSTER",
-    "name": { "str": "evolved mutant" },
+    "name": { "str": "bestial mutant" },
+    "//": "Midgame 1v1 splat, late is fiiine",
     "description": "A towering beast that is neither human nor animal anymore.  A malformed, huge creature covered in thick fur and the torn bottom part of a jumpsuit.  The sinister fangs, claws and the look of insanity in his pale yellow eyes are a testament to that he lost all of his humanity.",
```
[the remark this answers is not in the document]
Comment (Venera3) [reply data/json/monsters/mutant.json:52]: Whoops, I forgot to update the descriptions of both existing monsters.

File: data/json/monster_special_attacks/spells.json [hunk 2]
```diff
@@ -260,6 +260,36 @@
...
+    "max_range": 10,
+    "min_aoe": 8,
+    "max_aoe": 12,
+    "min_damage": 0,
+    "max_damage": 0,
+    "min_duration": 9000,
+    "max_duration": 9000,
+    "base_casting_time": 200,
+    "sound_description": "a screeching howl!",
+    "flags": [ "LOUD", "RANDOM_DAMAGE", "RANDOM_AOE" ]
```
[the remark this answers is not in the document]
Comment (Venera3) [reply data/json/monster_special_attacks/spells.json:291]: A relic from when it used `mod_speed` to speed up targets, good catch.

**Comment by Venera3** [comment]
Relatively happy with their nest - a few mattresses, the occasional canned food, the familiar field laboratory packed with mutagen ingredients... I also snuck in a minor visual fix (anesthesia machines looked like standing tanks, which is hilarious but very much wrong), and buffed the drug spawns in the hospital section.

**Comment by Venera3** [comment]
That's slightly worrying, I didn't hook up any `eoc`s for them and that looks like one of those errors.

**Pull request merged**